# Jawaban Kisi-Kisi LC 1


## koneksi Google Colaboratory dengan BigQuery

In [10]:
from google.colab import auth
from google.cloud import bigquery
auth.authenticate_user()
print('Authenticated')

project_id = "kisi-kisi-lc2-435411"
client = bigquery.Client(project=project_id)

Authenticated


In [ ]:
# Query SQL assign in 'query' variable
sql_query = """
SELECT *
  FROM ftds-hacktiv8-project.practice.bike_rent;
"""

In [ ]:
# Output would be saved in 'df' variable
df = client.query(sql_query).to_dataframe()

In [ ]:
df

trip_id subscriber_type bike_id bike_type                start_time  \
0     18896589        Explorer      93   classic 2018-10-28 14:16:14+00:00   
1     13948351        Local365     315   classic 2017-03-07 12:37:33+00:00   
2     16243803        Local365     877   classic 2017-10-21 09:49:48+00:00   
3     16222446         Walk Up     273   classic 2017-10-18 15:01:52+00:00   
4     16222451         Walk Up     201   classic 2017-10-18 15:02:26+00:00   
...        ...             ...     ...       ...                       ...   
7995  15469480         Walk Up     327   classic 2017-07-22 20:17:46+00:00   
7996  16130650         Walk Up     528   classic 2017-10-07 11:25:36+00:00   
7997  14204138         Walk Up     235   classic 2017-03-27 13:19:02+00:00   
7998  15037809         Walk Up     206   classic 2017-06-05 11:42:40+00:00   
7999   9499076         Walk Up     315   classic 2016-04-09 18:52:09+00:00   

                           start_station_name  \
0                     Dean Keeton & Speedway    
1                                 Main Office   
2           Republic Square @ 5th & Guadalupe   
3             Bullock Museum @ Congress & MLK   
4             Bullock Museum @ Congress & MLK   
...                                       ...   
7995          Bullock Museum @ Congress & MLK   
7996                  Congress & Cesar Chavez   
7997          State Capitol @ 14th & Colorado   
7998          State Capitol @ 14th & Colorado   
7999  Capital Metro HQ - East 5th at Broadway   

                             end_station_name  duration_minutes  
0                              11th & Salina                178  
1                                 Main Office               198  
2                                 Main Office                60  
3                                 Main Office                60  
4                                 Main Office                60  
...                                       ...               ...  
7995          Bullock Museum @ Congress & MLK                55  
7996          Bullock Museum @ Congress & MLK                55  
7997          State Capitol @ 14th & Colorado                55  
7998          State Capitol @ 14th & Colorado                55  
7999  Capital Metro HQ - East 5th at Broadway                55  

[8000 rows x 8 columns]

1. Berapa banyak label unique dari bike_type?

In [14]:
df = client.query("""
SELECT COUNT(DISTINCT (bike_type)) FROM ftds-hacktiv8-project.practice.bike_rent;
""").to_dataframe()

df

f0_
0    2

Ada 2 label unique dari kolom bike_type.

2. Apa saja nama label unique dari tipe sepeda?


In [13]:
df = client.query("""
SELECT DISTINCT (bike_type) FROM ftds-hacktiv8-project.practice.bike_rent;
""").to_dataframe()

df

bike_type
0   classic
1  electric

Nama label unique dari kolom bike_type adalah classic dan electric.

3. Berapa banyak total peminjaman sepeda pada tahun 2016 dan 2017?

In [21]:
query = """
SELECT COUNT(*) AS total_peminjaman
FROM ftds-hacktiv8-project.practice.bike_rent
WHERE EXTRACT(YEAR FROM TIMESTAMP(start_time)) BETWEEN 2016 AND 2017
"""

df = client.query(query).to_dataframe()
df

total_peminjaman
0              7993

Total peminjaman sepeda pada tahun 2016 dan 2017 adalah sebanyak 7993.